snpe/notebooks/non-marketplace_simulations.ipynb

```diff
@@ -0,0 +1,446 @@
+       "                        }\n",
+       "                        break;\n",
+       "                    }\n",
+       "                }\n",
+       "            }, 500);\n",
+       "            "
+      ],
+      "text/plain": [
+       "<IPython.core.display.Javascript object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "print(f\"The number of available CPUs is: {mp.cpu_count()}\")\n",
+    "torch.set_num_threads(mp.cpu_count())\n",
+    "print(f\"The number of CPUs to be employed will be {torch.get_num_threads()}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d74645eb",
+   "metadata": {},
+   "source": [
+    "### C. The simulation and its arguments"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c6ea6abe",
+   "metadata": {},
+   "source": [
+    "The generate and save function defined below, as you will see, is in charge of calling the desired simulator class and deliver the necessary arguments to run the simulation once we have provided them. Lets briefly review which are these arguments involved in the simulation.\n",
+    "\n",
+    "- __num_simulations__: Number of customers (Integer) to simulate. Each one is equivalent to the entire journey of a consumer consisting on:\n",
+    "    1. Reviewing the prior reviews of a product and forming an expectation of the product\n",
+    "    2. Purchase and experience the product\n",
+    "    3. Decide whether to leave a rating or not based on the difference between the expected experience and the actual experience. If the result of the decision is positive the consumer will continue to points 4 and 5, otherwise the simulation ends here.\n",
+    "    4. \"Decide\" whether to exhibit herding behavior in its rating in case it is being posted.\n",
+    "    5. Post the rating.\n",
+    "    \n",
+    "\n",
+    "- __review_prior__: Set of initial ratings of the product that are pre-loaded before the simulation starts, taking the shape of an array of five integer values. By default this is set as an array os five 1s. This implies that by the time the first consumer of the simulation reviews the product 5 prior reviews will be observed, each of them assigned to one of the five values composing the rating scale [1 - 5]. \n",
+    "\n",
+    "\n",
+    "- __tendency_to_rate__: Underliying tendency to rate for all consumers taking float values in the interval [0,1]. In other words, this is the proportion of consumers that will post a rating regardless of the value of the rho parameter(s) and the difference between their actual and expected product experience. If set at the default value of 0.05, 5% of all consumers will post a rating independently of the other factors at play in the simulation.\n",
+    "\n",
+    "\n",
+    "- __simulation_type__: Type of simulation output to produce between timeseries and histogram. Accepts the strings \"timeseries\" and \"histogram\" as inputs.\n",
```

Comment:
Let's add: Returns the timeseries of the simulated ratings, in a cumulative histogram format (so, the order of rating accumulation is preserved) if "timeseries" is chosen. For "histogram", returns the final histogram of ratings (and throws away the order of rating accumulation)

Reply:
Done!